core/src/impl/Kokkos_ViewArray.hpp

```diff
@@ -395,13 +395,13 @@ class ViewMapping<Traits, Kokkos::Array<> > {
     const size_t alloc_size =
         (m_impl_offset.span() * Array_N * MemorySpanSize + MemorySpanMask) &
         ~size_t(MemorySpanMask);
-
+    const std::string &alloc_name =
+        ((Kokkos::Impl::ViewCtorProp<void, std::string> const &)arg_prop).value;
```

Comment:
Avoid C-style cast.

Reply:
~I'm kind of confused about why this cast is necessary in the first place.~ `ViewCtorProp` is weird, and the presence of this cast raises a red flag for many readers. There should at least be a comment explaining why the cast is here and is like it is.

On further reflection here, I'd probably prefer to have a `get_as` or `get_argument` (or something) member function template in `ViewCtorProp` which make it clear what's going on here rather than having the cast everywhere that it's used and having to explain why there's a cast. I'm pretty okay at C++, know a lot about C++ patterns and idioms, and know Kokkos pretty well, and I had to look this up to figure out what was going on, so I suspect others will have issues also. (I can make this change if you need me to).

Reply:
also, please use `auto` here. `std::string` on the left-hand side of an assignment just screams accidental implicit conversion.